src/Model/Contact.php

```diff
@@ -1517,4 +1517,27 @@ public static function updateBirthdays()
 			}
 		}
 	}
+
+	/**
+	 * Remove the unavailable contact ids from the provided list
+	 *
+	 * @param array $contact_ids Contact id list
+	 */
+	public static function pruneUnavailable(array &$contact_ids)
+	{
+		if (empty($contact_ids)) {
+			return;
+		}
+
+		$str = dbesc(implode(',', $contact_ids));
+
+		$stmt = dba::p("SELECT `id` FROM `contact` WHERE `id` IN ( " . $str . ") AND `blocked` = 0 AND `pending` = 0 AND `archive` = 0");
```

Comment:
When PR https://github.com/friendica/friendica/pull/4532 is accepted, this can be written as:
```
$condition = ['id' => $contact_ids, 'blocked' => false, 'pending' => false, 'archive' => false];
$stmt = dba::select('contact', $condition);
```
In fact I wrote that PR because I saw this query ;-)

Reply:
I'm not ready to merge it just yet, see my comment.

Reply:
You don't need to change it here. I just wanted to point out that we now have this possibility.

Reply:
Great, I like this idea!